src/main.rs

```diff
@@ -99,22 +126,68 @@ fn main() {
+        Some(format_type_str) => format_type_str
+            .parse::<FormatType>()
+            .expect("--format-type must be one of the following: [hex|bip39]"),
+        None => FormatType::Hex,
+    };
+
+    let format_error = || panic!("bip39 encoding is only available for 128 and 256 bit keys");
+
     match bitsize {
-        8 => dispatch_shamir_type::<GF8>(matches, threshold, shares),
-        16 => dispatch_shamir_type::<GF16>(matches, threshold, shares),
-        32 => dispatch_shamir_type::<GF32>(matches, threshold, shares),
-        64 => dispatch_shamir_type::<GF64>(matches, threshold, shares),
-        128 => dispatch_shamir_type::<GF128>(matches, threshold, shares),
-        256 => dispatch_shamir_type::<GF256>(matches, threshold, shares),
+        8 => match format_type {
+            FormatType::Hex => dispatch_shamir_type::<GF8>(matches, threshold, shares, format_type),
+            FormatType::Bip39 => format_error()
+        },
+        16 => match format_type {
+            FormatType::Hex => dispatch_shamir_type::<GF16>(matches, threshold, shares, format_type),
+            FormatType::Bip39 => format_error()
+        },
+        32 => match format_type {
+            FormatType::Hex => dispatch_shamir_type::<GF32>(matches, threshold, shares, format_type),
+            FormatType::Bip39 => format_error()
+        },
+        64 => match format_type {
+            FormatType::Hex => dispatch_shamir_type::<GF64>(matches, threshold, shares, format_type),
+            FormatType::Bip39 => format_error()
+        },
+        128 => match format_type {
+            FormatType::Hex => dispatch_shamir_type::<GF128>(matches, threshold, shares, format_type),
+            FormatType::Bip39 => dispatch_mnemonic_shamir_type::<GF128>(matches, threshold, shares, format_type),
+        },
+        256 => match format_type {
+            FormatType::Hex => dispatch_shamir_type::<GF256>(matches, threshold, shares, format_type),
+            FormatType::Bip39 => dispatch_mnemonic_shamir_type::<Bip39<GF256>>(matches, threshold, shares, format_type),
+        },
         _ => panic!("Unsupported bitsize: {}", bitsize),
     }
 }
 
-fn dispatch_shamir_type<F: Field + Debug + Display>(matches: ArgMatches, k: usize, n: usize) {
+fn dispatch_shamir_type<F: Field + Debug + Display>(
+    matches: ArgMatches,
+    k: usize,
+    n: usize,
+    format_type: FormatType
+) {
     let shamir_type = matches.value_of("type").unwrap();
     match shamir_type {
-        "compact" => process_command::<F, CompactShamir>(matches, k, n),
-        "random" => process_command::<F, RandomShamir>(matches, k, n),
+        "compact" => process_command::<F, CompactShamir>(matches, k, n, format_type),
+        "random" => process_command::<F, RandomShamir>(matches, k, n, format_type),
+        _ => panic!("Unsupported shamir type: {}", shamir_type),
+    };
+}
+
+fn dispatch_mnemonic_shamir_type<F: Field + Debug + Display>(
```

Comment:
It seems redundant to me to have 2 separate functions for the 2 `FormatType`s, but that each also take a `FormatType` parameter.